src/View/ViewVarsTrait.php

```diff
@@ -65,6 +66,7 @@ public function getView($viewClass = null)
         }
 
         if ($this->View && $this->View instanceof $className) {
+            $this->View->viewVars = $this->viewVars;
```

Comment:
@markstory Can't easy to sync rest of the properties. For eg. the helper instance would have been already generated based on helpers list passed to view constructor. Rest of the properties like `$view`, `$layout` i have already taken care of in 3.1 by using magic methods in controller to directly change the view class properties.